oci/oci.go

```diff
@@ -32,35 +32,47 @@ func SetCapabilities(s *specs.Spec, caplist []string) error {
 	return nil
 }
 
-// AppendDevicePermissionsFromCgroupRules takes rules for the devices cgroup to append to the default set
-func AppendDevicePermissionsFromCgroupRules(devPermissions []specs.LinuxDeviceCgroup, rules []string) ([]specs.LinuxDeviceCgroup, error) {
+// DevicePermissionsFromCgroupRules takes rules for the devices cgroup
+func DevicePermissionsFromCgroupRules(rules []string) ([]specs.LinuxDeviceCgroup, error) {
+	var devPermissions []specs.LinuxDeviceCgroup
 	for _, deviceCgroupRule := range rules {
 		ss := deviceCgroupRuleRegex.FindAllStringSubmatch(deviceCgroupRule, -1)
 		if len(ss) == 0 || len(ss[0]) != 5 {
 			return nil, fmt.Errorf("invalid device cgroup rule format: '%s'", deviceCgroupRule)
 		}
 		matches := ss[0]
 
+		if matches[0] == "a" || matches[1] == "a" {
+			ms := deviceCgroupARuleRegex.MatchString(matches[0])
+			if !ms {
+				return nil, fmt.Errorf("although this cgroup rule is technically correct, because 'a' maps to 'a *:* rwm' regardless of what comes next, this format is partially ineffective: '%s'", deviceCgroupRule)
+			}
+
+			// TODO Is the type field needed?
+			dPermissions := specs.LinuxDeviceCgroup{
+				Allow:  true,
+				Type:   "a",
+				Access: "rwm",
+			}
+
+			devPermissions = append(devPermissions, dPermissions)
+			return devPermissions, nil
+		}
+
 		dPermissions := specs.LinuxDeviceCgroup{
 			Allow:  true,
 			Type:   matches[1],
 			Access: matches[4],
 		}
-		if matches[2] == "*" {
-			major := int64(-1)
```

Comment:
w.r.t. Major/Minor `nil` being equivalent to `-1` (runtime-spec); Looks indeed that runc uses these as a default, and only override in the "not nil" situation; https://github.com/opencontainers/runc/blob/ee06c700dcb34504204cb171bb35e0e01477bf58/libcontainer/specconv/spec_linux.go#L475-L487

Should we check if other runtimes (e.g. `crun` handle it the same?)